### PR TITLE
Allow Badge's variant prop to take non-semantic colors as well

### PR DESCRIPTION
**Pull request by parichay28**
### Description 

- PG and X to have separate color mappings for `tertiary` (cyan and xCyan respectively)
- Badge now accepts non-semantic variant `cyan`

### CODE REVIEW

File: packages/blade-old/src/atoms/Badge/Badge.native.js [hunk 1]
```diff
@@ -38,9 +38,9 @@ const styles = {
         return getColor(theme, 'primary.900');
       default:
         if (fill === 'subtle') {
-          return getColor(theme, 'positive.930');
+          return variant ? getColor(theme, variant) : getColor(theme, 'positive.930');
```
Comment (sumit-gupta91) [inline packages/blade-old/src/atoms/Badge/Badge.native.js:41]: Once we start passing colors as variant, how will we make sure people are not passing colors which are not in theme, also does getColor function has a default case color ?

**Review by sumit-gupta91 — CHANGES_REQUESTED** [review]

File: packages/blade-old/src/atoms/Badge/Badge.native.js (hunk 1 above)
Comment (parichay28) [reply packages/blade-old/src/atoms/Badge/Badge.native.js:41]: getColor throws error on wrong color name

File: packages/blade-old/src/atoms/Badge/Badge.native.js [hunk 2]
```diff
@@ -62,6 +67,8 @@ const styles = {
           return 'neutral.900';
         case 'neutral':
           return 'primary.900';
+        case 'cyan':
+          return 'tertiary.900';
```
Comment (chaitanyadeorukhkar) [inline packages/blade-old/src/atoms/Badge/Badge.native.js:71]: If variant is `cyan` and `fill` is `solid`, you'll end up having background as well as foreground as `tertiary.900`, right?

Comment (parichay28) [reply packages/blade-old/src/atoms/Badge/Badge.native.js:71]: In that case, it'll be `primary.900`
image

**Review by chaitanyadeorukhkar — APPROVED** [review]

**Review by chaitanyadeorukhkar — CHANGES_REQUESTED** [review]
Can you update the tests as well?

https://github.com/razorpay/blade/blob/add%2Fbadge-non-semantic-colors/packages/blade-old/src/atoms/Badge/__tests__/Badge.native.test.js

**Review by chaitanyadeorukhkar — APPROVED** [review]

**Review by sumit-gupta91 — APPROVED** [review]

**Pull request merged**